dvc/config.py

```diff
@@ -119,6 +119,17 @@ class RelPath(str):
     "password": str,
     "ask_password": Bool,
 }
+WEBDAV_COMMON = {
+    "user": str,
+    "password": str,
+    "ask_password": Bool,
+    "root": str,
```

Comment:
You probably don't need `root` here, as that `url` from `REMOTE_COMMON should work, no?

Reply:
Yes, probably it is not necessary any more, I used it to specify the location of the WebDAV API endpoint (e.g. `/public.php/webdav`). But I still have a bit of confusion there...

Since I now terminate the `makedirs` at the `path` part of the initial `path_info` (which comes from the `url` of `REMOTE_COMMON`), the `root` option might be removed to not confuse others as well.

I somehow still wonder if there could be a valid use case for the option... At least there must be a reason for `webdavclient3` to offer this option?

Reply:
Cannot we split that into `hostname` and `root` from the `url`:
Eg:  `http://owncloud.com/remote.php/webdav` -> `http://owncloud.com` and `/remote.php/webdav`?